runtime/java/src/common/Prodleton.java

```diff
@@ -0,0 +1,19 @@
+package common;
+
+
+public abstract class Prodleton<T> {
```

Comment:
Would be good to add a javadoc comment for this class explaining what a "Prodleton" is (and why the name...)  

Also, indentation in this file seems messed up.